runner:
    stabilization_timeout: 0.25
    acceleration_timeout: 0.5
    verification_timeout: 0.25
    decceleration_timeout: 4
    goal_reach_timeout: 10
    emergency_wait_timeout: 3.5

interpreters:
  velocity_interpreter_node: # node name
    max_vel: 0.8
    rate: 100

calculators:
  robot_type: "turtlebot3_sim"
  test_type: "test1_braking"

  test_comp_param_node: # node name 
    param: 0.0
    tolerance: 0.5

  test_comp_poses_node: # node name 
    pose_tolerance: 0.0
    orientation_tolerance: None
  
  test_comp_topics_node: # node name 
    tolerance: 0

  test_folder_bagger_node:
    start_offset: 2 # Wait these many seconds before killing
    stop_offset: 2 # Wait these many seconds before killing
    whitelist:
    # Default whitelist: ["/signal/", "/data", "usb_cam", "/tf"]
    # Any topics which have one of these STRINGS in them will be added to the whitelist for recording,
      - hedge
      - ros
    blacklist:
    # Any WHITELISTED topics which have one of these STRINGS in them will be rejected for recording
      - compressed

  rosparam_logger_node:
    start_offset: 1
    stop_offset: 0.5
    namespaces:
      - "runner"
      - "interpreters"
      - "calculators"

  results_logger_node:
    start_offset: 0.5
    stop_offset: 0.1
    logginng_topics:
      - /data/calc/braking_time
      - /data/calc/braking_distance

  speed_logger:
    start_offset: 1.5 # Must be longer than folder bagger, since this tends to be created first.
    stop_offset: 0.1
    logged_data_title: "robot_speed"
    column_headers: ["Time", "Speed"]

  robot_pose_logger:
    start_offset: 1.5 # Must be longer than folder bagger, since this tends to be created first.
    stop_offset: 0.1
    logged_data_title: "robot_pose"
    column_headers: ["Time", "Seq",
                              "Pose.Position.x",
                              "Pose.Position.y",
                              "Pose.Position.z",
                              "Pose.Orientation.x",
                              "Pose.Orientation.y",
                              "Pose.Orientation.z",
                              "Pose.Orientation.w",
                              "Roll(deg)", "Pitch(deg)", "Yaw(deg)"]
